### PR TITLE
Observer pods

### PR DESCRIPTION
**Pull request by jmguzik**
/cc @jmguzik 
/hold

### CODE REVIEW

**Comment by jmguzik** [comment]
As discussed today, for now, temporarily e2e test will be removed (I removed file from the PR, and left the rest of the files) and it will be the task for the next sprint to implement it properly.

**Comment by jmguzik** [comment]
/cc @danilo-gemoli @bbguimaraes 
were involved in the discussion.

**Comment by jmguzik** [comment]
As discussed with @bear-redhat we should not care so much about deps.
Not sure is optional or not so:
/override ci/prow/snyk-deps

**Comment by danilo-gemoli** [comment]
I hope this will not break anything
/lgtm

**Comment by jmguzik** [comment]
> I hope this won't break anything /lgtm

My hope is the same xD.

File: test/e2e/observer/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
+//go:build e2e
```
Comment (jmguzik) [inline test/e2e/observer/e2e_test.go]: This file was removed and should be re-added once we figure out e2e testing.

**Pull request merged**